assets/js/modules/analytics-4/datastore/webdatastreams.js

```diff
@@ -310,6 +310,33 @@ const baseSelectors = {
 		}
 	),
 
+	/**
+	 * Gets matched web data stream for selected property.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state      Data store's state.
+	 * @param {string} propertyID The GA4 property ID to find matched web data stream.
+	 * @return {(Object|null|undefined)} A web data stream object if found, otherwise null; `undefined` if web data streams are not loaded.
+	 */
+	getMatchingWebDataStreamByPropertyID: createRegistrySelector(
+		( select ) => ( _state, propertyID ) => {
+			const datastreams =
+				select( MODULES_ANALYTICS_4 ).getWebDataStreams( propertyID );
+
+			if ( datastreams === undefined ) {
+				return undefined;
+			}
+
+			const matchingDataStream =
+				select( MODULES_ANALYTICS_4 ).getMatchingWebDataStream(
+					datastreams
+				);
+
+			return matchingDataStream || null;
```

Comment:
This catches my eye as not quite right thinking that `getMatchingWebDataStream` might return `undefined` in which case it would be incorrect to return `undefined || null`. We should always return `undefined` until we "know" there isn't something to return, then we return `null`.

In this case, the selector only ever returns `Object|null` so this fallback is redundant.

Reply:
Understood, thank you for the context and clarification!